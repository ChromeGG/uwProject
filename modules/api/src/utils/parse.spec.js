const { readFileSync } = require('fs')
const { parseHtmlDocument } = require('./parse')

describe('parseHtmlDocument', () => {
  test('can parse basic html document', async () => {
    const result = parseHtmlDocument(
      'dane.html',
      readFileSync(`${__dirname}/../../tests/fixtures/dane.html`).toString()
    )

    expect(result).toMatchSnapshot()
  })
})
