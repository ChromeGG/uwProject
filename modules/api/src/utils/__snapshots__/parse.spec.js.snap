// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseHtmlDocument can parse basic html document 1`] = `
Object {
  "balance": 1263.46,
  "charges": Array [
    Object {
      "date": "2019-10-20",
      "details": "PARZYCMIECHY",
      "operation": "ZAKUPY WODY",
      "value": -21.01,
    },
    Object {
      "date": "2019-10-20",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP CHLEBA",
      "value": -24.68,
    },
    Object {
      "date": "2019-10-20",
      "details": "ZAKOPANE",
      "operation": "ZAKUP GUMY ORBIT",
      "value": -5,
    },
    Object {
      "date": "2019-10-20",
      "details": "CZĘSTOCHOWA",
      "operation": "ZAKUP CHRUPEK",
      "value": -2.91,
    },
    Object {
      "date": "2019-10-21",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP KSIAZKI",
      "value": -22.65,
    },
    Object {
      "date": "2019-10-21",
      "details": "SOPOT",
      "operation": "ZAKUP GRZEBIENIA",
      "value": -5.77,
    },
    Object {
      "date": "2019-10-22",
      "details": "KRAKOW",
      "operation": "ZAKUP 20KG MARCHEWKI",
      "value": -212.17,
    },
    Object {
      "date": "2019-10-22",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP MASŁA",
      "value": -3.19,
    },
    Object {
      "date": "2019-10-23",
      "details": "PARZYCMIECHY",
      "operation": "BILETY DO KINA",
      "value": -15.49,
    },
    Object {
      "date": "2019-10-23",
      "details": "CZESTOCHOWA",
      "operation": "KFC",
      "value": -16.18,
    },
    Object {
      "date": "2019-10-24",
      "details": "WARSZAWA",
      "operation": "ZAKUP 20 HOD-DOG",
      "value": -74.23,
    },
    Object {
      "date": "2019-10-24",
      "details": "TARNÓW",
      "operation": "ZAKUP GAZETY",
      "value": -10.67,
    },
    Object {
      "date": "2019-10-24",
      "details": "WARSZAWA",
      "operation": "ZAKUP PIETRUSZKI",
      "value": -15.9,
    },
    Object {
      "date": "2019-10-25",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP PIWA",
      "value": -18.13,
    },
    Object {
      "date": "2019-10-25",
      "details": "SOPOT",
      "operation": "ZAKUP OPON",
      "value": -10.13,
    },
    Object {
      "date": "2019-10-25",
      "details": "WARSZAWA",
      "operation": "ZAKUP GRY",
      "value": -114.52,
    },
    Object {
      "date": "2019-10-25",
      "details": "GDYNIA",
      "operation": "ZAKUP TELEWIZORA",
      "value": -155.25,
    },
    Object {
      "date": "2019-10-26",
      "details": "MSZANA",
      "operation": "ZAKUP FRYTEK",
      "value": -35.93,
    },
    Object {
      "date": "2019-10-26",
      "details": "PARZYCMIECHY",
      "operation": "ZAKUP BILETU",
      "value": -3.09,
    },
    Object {
      "date": "2019-10-26",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUPY-OBIAD W RESTAURACJI",
      "value": -55.89,
    },
    Object {
      "date": "2019-10-26",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP CZAPKI",
      "value": -42.58,
    },
    Object {
      "date": "2019-10-26",
      "details": "GOOGLE",
      "operation": "ZAKUP PROGRAMU TELEWIZYJNEGO",
      "value": -7.49,
    },
    Object {
      "date": "2019-10-26",
      "details": "OSTRAWA",
      "operation": "ZAKUP BILETU DO POCIĄGU",
      "value": -106.2,
    },
    Object {
      "date": "2019-10-27",
      "details": "CZESTOCHOWA BP-GROSIK 321",
      "operation": "ZAKUP PRZY U�YCIU KARTY W KRAJU",
      "value": -66.96,
    },
    Object {
      "date": "2019-10-27",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP SPARPETEK",
      "value": -30.06,
    },
    Object {
      "date": "2019-10-27",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUP OBIADU",
      "value": -73.47,
    },
    Object {
      "date": "2019-10-27",
      "details": "WARSZAWA",
      "operation": "ZAKUP TOREBKI",
      "value": -9.99,
    },
    Object {
      "date": "2019-10-27",
      "details": "TARNÓW",
      "operation": "ZAKUP PROGRAMU DO TWORZENIA ZDJĘĆ",
      "value": -93.42,
    },
    Object {
      "date": "2019-10-27",
      "details": "CZESTOCHOWA",
      "operation": "ZAKUPY- PIZZA",
      "value": -10.5,
    },
  ],
  "fileName": "dane.html",
  "recognition": Array [
    Object {
      "date": "2019-10-29",
      "details": "",
      "operation": "SP�ATA - PRZELEW WEWN�TRZNY - DZI�KUJEMY",
      "value": 0,
    },
  ],
}
`;
